pkg/operator/operator.go

```diff
@@ -489,6 +489,18 @@ func (o *Operator) sync(key string) error {
 	return nil
 }
 
+func (o *Operator) reportError(err error, taskName string) {
+	if o.failedReconcileAttempts > 3 {
+		klog.Infof("Updating ClusterOperator status to failed after retrying %v times. Err: %v", o.failedReconcileAttempts, err)
```

Comment:
I might be worth logging the error message at every step? IIUC the first 3 errors would be missing currently.